### PR TITLE
iOS native player

### PR DESCRIPTION
**Pull request by djinely**

### CODE REVIEW

File: ios/KalturaPlayerViewManager.swift [hunk 1]
```diff
@@ -299,14 +326,6 @@ class KalturaPlayerViewManager: RCTViewManager {
...
             KalturaPlayerEvents.emitter.sendEvent(withName: "bookmarkError", body: event.data)
         }
-        self.kalturaPlayer.addObserver(self, event: OttEvent.concurrency) { event in
-            KalturaPlayerEvents.emitter.sendEvent(withName: "loadMediaFailed", body: [
-                "code": "ConcurrencyLimitation",
-                "message": "Concurrency limitation",
-                "name": "OTTError",
-                "type": "loadMediaFailed"
-            ])
-        }
```
Comment (OrenMe) [inline ios/KalturaPlayerViewManager.swift]: U need to keep this and emit a concurrency error like in android

File: ios/KalturaPlayerViewManager.swift [hunk 2]
```diff
@@ -384,6 +403,19 @@ class KalturaPlayerRNView : UIView {
         kalturaPlayer.loadMedia(options: mediaOptions) { error in
             if (error != nil) {
                 print("Error in loadMedia: %@", error!)
+                KalturaPlayerEvents.emitter.sendEvent(withName: "loadMediaFailed", body: [
+                    "code": "ConcurrencyLimitation",
+                    "message": "Concurrency limitation",
+                    "name": "OTTError",
```
Comment (OrenMe) [inline ios/KalturaPlayerViewManager.swift]: this shouldnt be hardcoded, you need to get the error from the player and just make sure the object has the same structure like JS layer expects

File: ios/KalturaPlayerViewManager.swift [hunk 3]
```diff
@@ -384,6 +403,19 @@ class KalturaPlayerRNView : UIView {
...
+                    "type": "loadMediaFailed"
+                ])
+            } else {
+                KalturaPlayerEvents.emitter.sendEvent(withName: "loadMediaSuccess", body: [
+                    "payload": [
+                        "id": assetId,
+                        "type": "loadMediaFailed"
```
Comment (OrenMe) [inline ios/KalturaPlayerViewManager.swift]: ?!?!? loadMediaSuccess --->loadMediaFailed
and also here this is thr same object as we get from Android?
I know we are not getting the same object from `loadMedia` completion block but we have apis in player to get the MediaAsset object.
Please compare android and iOS and expose same data!

File: ios/KalturaPlayerViewManager.swift [hunk 4]
```diff
@@ -326,6 +326,14 @@ class KalturaPlayerViewManager: RCTViewManager {
         self.kalturaPlayer.addObserver(self, event: OttEvent.bookmarkError) { event in
             KalturaPlayerEvents.emitter.sendEvent(withName: "bookmarkError", body: event.data)
         }
+        self.kalturaPlayer.addObserver(self, event: OttEvent.concurrency) { event in
+            KalturaPlayerEvents.emitter.sendEvent(withName: "loadMediaFailed", body: [
```
Comment (OrenMe) [inline ios/KalturaPlayerViewManager.swift]: look in android, the event name is not correct

**Comment by giladna** [comment]
seems not relevant

**Pull request closed without merging**